webui/react/src/types.ts

```diff
@@ -163,21 +165,39 @@ export enum TaskType {
   Tensorboard = 'TENSORBOARD',
 }
 
-export interface RecentTask {
+export interface Task {
   archived?: boolean;
+  // source?: Record<TBSourceType, number[]>;
+  misc?: CommandMisc;
   title: string;
   type: TaskType;
-  lastEvent: {
-    name: string;
-    date: string;
-  };
   id: string;
   ownerId: number;
   progress?: number;
   url?: string;
+  username?: string;
+  startTime: string;
   state: RunState | CommandState;
 }
 
+// food for thought for future work
+// remove exlusive fields from Task and:
+// export interface ExperimentTask extends Task {
+//   progress?: number;
+//   archived?: boolean;
+// }
+
+// export interface CommandTask extends Task {
+//   misc?: CommandMisc;
+// }
```

Comment:
thought: This is a great idea.
Recently had a PR that had this similar theme of extending more specifications (can't remember what  specifically...), but ran into some TypeScript issues when handling the functions that dealt with the extended types. Might need to just try it out and we where it lands.

Reply:
i'll remove and leave this for future.

Reply:
it will touch a lot of different parts. the swagger generated ts code is gonna give us some types we could use instead of the ones we have defined.